test/state/precompiles.cpp

```diff
@@ -364,19 +364,16 @@ evmc::Result call_precompile(evmc_revision rev, const evmc_message& msg) noexcep
     if (gas_left < 0)
         return evmc::Result{EVMC_OUT_OF_GAS};
 
-    // Buffer for the precompile's output.
-    // Big enough to handle all "expmod" tests, but in case does not match the size requirement
-    // from analysis, the result will likely be incorrect.
-    // TODO: Replace with std::pmr::monotonic_buffer_resource?
-    uint8_t output_buf[4096];
-    assert(std::size(output_buf) >= max_output_size);
-
+    const auto output_buf = static_cast<uint8_t*>(std::malloc(max_output_size));
```

Comment:
Wouldn't `new uint8_t[max_output_size]` be better, at least in that it doesn't require a cast?